src/service/world-api-adapter.js

```diff
@@ -192,10 +192,59 @@ module.exports = function (config) {
         /**
         * Add a user or list of users to a given world
         *
+        * Supported format for users
+        * wm.addUsers('123-123-123-123');
+        * wm.addUsers({ userId: '123-123-123-123', role: 'abc' });
+        * wm.addUsers(['123-123-123-123', '312-321-321-321']);
+        * wm.addUsers([{ userId: '123-123-123-123', role: 'abc' }, { userId: '312-321-321-321' }]);
+        *
+        * note that options can be passed as the second parameter, so these calls are both valid
+        *   wm.addUser('123-123-123', 'game1')
+        *   wm.addUser('123-123-123', { filter: game1 })
+        *
+        * @param users {string|object|array} the users to add to the world
+        * @param worldId {string} (optional) the worldId to add the users to.
+        *       If not specified it will take the filter paramter of the options or the filter populated from previous calls
+        * @param options {object} (optional)
         */
-        addUsers: function (users, options) {
+        addUsers: function (users, worldId, options) {
+
```

Comment:
I think you can combine the isString checks so you don't have to do it once outside the isArray and once inside. Maybe something like:

```javascript
users = $.map([].concat(users), function (u){
   return $.isPlainObject(u) ? u : {userId : u};
});
```